include/boost/gil/image_processing/threshold.hpp

```diff
@@ -95,16 +121,18 @@ void threshold_binary
     }
 }
 
-/*!
-Takes an image view and performes binary thresholding operation on each chennel.
-If direction is regular:
-values more or equal to threshold_value will be set to maximum numeric limit of channel else 0
-If direction is inverse:
-values more than or equal to threshold_value will be set to 0 else maximum numeric limit of channel
-*/
+/// \ingroup ImageProcessing
+/// \brief Applies fixed threshold to each pixel of image view.
+/// Performs image binarization by thresholding channel value of each
+/// pixel of given image view.
+/// This variant of threshold_binary automatically deduces maximum value for each channel
+/// of pixel based on channel type.
+/// If direction is regular:
+/// values more or equal to threshold_value will be set to maximum numeric limit of channel else 0
```

Comment:
After merging #321 `threshold_value` is treated as a boundary so it is not included into valid range. 
Only pixel values more than `threshold_value` will be set to the maximum numeric limit.

Reply:
Done